interpreter.go

```diff
@@ -422,19 +311,61 @@ func (i *interpreter) evaluate(a astNode) (value, error) {
 				// Omitted field.
 				continue
 			default:
-				return nil, makeRuntimeError(ast.Loc(), "Field name was not a string.")
+				return nil, e.Error("Field name was not a string.")
 			}
 
 			if err != nil {
```

Comment:
You sure? Maybe I'm not seeing something, but I've added (not pushed yet) a test that actually gets there...